src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts

```diff
@@ -193,6 +193,13 @@ export class InlineChatController implements IEditorContribution {
 
 	// ---- state machine
 
+	private _showWidget(initialRender: boolean = false, response?: SessionResponse) {
+		assertType(this._activeSession);
+		const selectionRange = this._activeSession.wholeRange.value;
+		const widgetPosition = this._strategy?.getWidgetPosition(initialRender, selectionRange, response);
```

Comment:
Looks like all implementations for `initialRender === true` are the same. How about not giving strategies that choice and simplying using the cursor position when doing the initial render, e.g the strategies are only called when `initialRender` is `false` and the initial case is handled here. Btw - to get cursor position you can simply call `ICodeEditor.getPosition()` - no need to use the view model etc

Reply:
I implemented the change from the review below, the initial cursor position is saved in the constructor. It is perhaps cleaner in that case to keep the initialRender variable, since we can directly return this._initialPosition when initialRender is true